### PR TITLE
Use `NetworkStream` from CoreFX.

### PR DESCRIPTION
**Pull request by baulig**
This brings the `NetworkStream` class and related tests from CoreFX.

* `SocketAsyncEventArgs`: partially replace the following properties with the CoreFX version:
  `Buffer`, `BufferList`, `Offset`, `SetBuffer ()` as well as the internal `CopyBufferFrom ()`.

* `SocketAsyncResult`: change `Buffer` from `byte[]` to `Memory<byte>`.

* Add internal `Socket` overloads:
   - `Receive (Memory<byte> buffer, int, int, SocketFlags, out SocketError)`
   - `ReceiveFrom (Memory<byte>, int, int, SocketFlags, ref EndPoint, out SocketError)`
   - `SendTo (Memory<byte>, int, int, SocketFlags, EndPoint)`

* Implement async task overloads from CoreFX (`Socket.Tasks.cs`) to `Socket`:
   - `int Read(Span<byte>)`
   - `Task<int> ReadAsync(byte[], int, int, CancellationToken)`
   - `ValueTask<int> ReadAsync(Memory<byte>, CancellationToken)`
   - `int ReadByte()`
   - `void Write(ReadOnlySpan<byte>)`
   - `Task WriteAsync(byte[], int, int, CancellationToken)`
   - `ValueTask WriteAsync(ReadOnlyMemory<byte>, CancellationToken)`
   - `void WriteByte(byte)`

* remove internal unused `PooledStream` and `ConnectionPool` APIs.

* bring new tests from CoreFX:
  - `System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs`
  - `System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.netcoreapp.cs`

* corlib: add `System.IO.FileSystem` and `System.Net.Sockets` resources and update SR.

### CODE REVIEW

File: mcs/class/System/System.Net.Sockets/Socket.cs [hunk 1]
```diff
@@ -1709,6 +1732,45 @@ internal int ReceiveFrom (byte [] buffer, int offset, int size, SocketFlags sock
...
 
+		int ReceiveFrom (Memory<byte> buffer, int offset, int size, SocketFlags socketFlags, ref EndPoint remoteEP, out SocketError errorCode)
+		{
+			SocketAddress sockaddr = remoteEP.Serialize();
+
+			int nativeError;
+			int cnt;
+			unsafe {
+				using (var handle = buffer.Slice (offset, size).Pin ()) {
+					Console.Error.WriteLine ($"RECEIVE FROM MEMORY!");
```
Comment (marek-safar) [inline mcs/class/System/System.Net.Sockets/Socket.cs]: Some leftover?

File: mcs/class/corlib/Makefile [hunk 2]
```diff
@@ -37,11 +37,13 @@ RESX_RESOURCE_STRING = \
 	../../../external/corefx/src/System.Collections.NonGeneric/src/Resources/Strings.resx \
 	../../../external/corefx/src/System.ComponentModel.TypeConverter/src/Resources/Strings.resx \
 	../../../external/corefx/src/System.Buffers/src/Resources/Strings.resx \
+	../../../external/corefx/src/System.IO.FileSystem/src/Resources/Strings.resx \
```
Comment (marek-safar) [inline mcs/class/corlib/Makefile]: Why do we need them in mscorlib?

File: mcs/class/System/System.Net.Sockets/Socket.cs (hunk 1 above)
Comment (baulig) [reply mcs/class/System/System.Net.Sockets/Socket.cs]: fixed.

File: mcs/class/corlib/Makefile (hunk 2 above)
Comment (baulig) [reply mcs/class/corlib/Makefile]: Somebody manually edited the SR files instead of adding them to the `RESX_RESOURCE_STRING`, so I had to fix that.

**Comment by steveisok** [comment]
@monojenkins commit csproj

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Pull request merged**